llama.cpp

```diff
@@ -7360,6 +7360,61 @@ void llama_sample_top_p(struct llama_context * ctx, llama_token_data_array * can
     }
 }
 
+void llama_sample_min_p(struct llama_context * ctx, llama_token_data_array * candidates, float p, size_t min_keep) {
```

Comment:
Nice progress with the refactoring. Is `p` here supposed to be state that's held/gets updated when the sampler is called (I assumed that was the case since the initial implementation read/wrote a file) or is it just a constant setting?

If it's state, then this would need to be `float * p`, however I don't see you updating `p` or anything so maybe it's just a static setting?

Reply:
p here represents the statically held Min P value which is now passed via `--min-p`, the function truncates logits based on this value. reading/writing a value was merely convenient for testing because i didn't need to have support in any GUI to get it working with kobold's api.

Reply:
My mistake, you can disregard the stuff I said earlier about using Mirostat state as a reference. Seems like you're now most of the way there then.